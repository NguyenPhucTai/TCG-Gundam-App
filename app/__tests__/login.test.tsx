import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react-native';
import { Alert } from 'react-native';
import LoginScreen from '../login';

// Mock Alert.alert
jest.spyOn(Alert, 'alert').mockImplementation(() => {});

describe('LoginScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders login form correctly', () => {
    render(<LoginScreen />);
    
    expect(screen.getByTestId('login-title')).toBeTruthy();
    expect(screen.getByText('Chào mừng bạn quay lại! Hãy tiếp tục hành trình của bạn.')).toBeTruthy();
    expect(screen.getByTestId('email-input')).toBeTruthy();
    expect(screen.getByTestId('password-input')).toBeTruthy();
  });

  it('has all required buttons', () => {
    render(<LoginScreen />);
    
    expect(screen.getByTestId('login-button')).toBeTruthy();
    expect(screen.getByTestId('signup-button')).toBeTruthy();
    expect(screen.getByText('Bỏ qua ngay bây giờ')).toBeTruthy();
  });

  it('allows text input in email and password fields', () => {
    render(<LoginScreen />);
    
    const emailInput = screen.getByTestId('email-input');
    const passwordInput = screen.getByTestId('password-input');
    
    fireEvent.changeText(emailInput, 'test@example.com');
    fireEvent.changeText(passwordInput, 'password123');
    
    expect(emailInput.props.value).toBe('test@example.com');
    expect(passwordInput.props.value).toBe('password123');
  });

  it('shows loading state when login button is pressed', async () => {
    render(<LoginScreen />);
    
    const emailInput = screen.getByTestId('email-input');
    const passwordInput = screen.getByTestId('password-input');
    const loginButton = screen.getByTestId('login-button');
    
    fireEvent.changeText(emailInput, 'test@example.com');
    fireEvent.changeText(passwordInput, 'password123');
    fireEvent.press(loginButton);
    
    await waitFor(() => {
      expect(screen.getByText('Đang đăng nhập...')).toBeTruthy();
    });
  });

  it('shows error when fields are empty', () => {
    render(<LoginScreen />);
    
    const loginButton = screen.getByTestId('login-button');
    fireEvent.press(loginButton);
    
    expect(Alert.alert).toHaveBeenCalledWith('Lỗi', 'Vui lòng nhập email và mật khẩu');
  });

  it('matches snapshot', () => {
    const tree = render(<LoginScreen />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
