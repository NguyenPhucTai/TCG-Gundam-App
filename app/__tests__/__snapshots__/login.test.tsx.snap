// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoginScreen matches snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "backgroundColor": "#F8F9FA",
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 40,
        }
      }
    >
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 32,
            "fontWeight": "bold",
            "marginBottom": 8,
          }
        }
        testID="login-title"
      >
        Đăng nhập
      </Text>
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Chào mừng bạn quay lại! Hãy tiếp tục hành trình của bạn.
      </Text>
    </View>
    <View
      style={
        {
          "marginBottom": 32,
        }
      }
    >
      <View
        style={
          {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
              "fontWeight": "600",
              "marginBottom": 8,
            }
          }
        >
          Email
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Nhập email của bạn"
          style={
            {
              "backgroundColor": "white",
              "borderColor": "#E5E5EA",
              "borderRadius": 12,
              "borderWidth": 1,
              "fontSize": 16,
              "paddingHorizontal": 16,
              "paddingVertical": 14,
            }
          }
          testID="email-input"
          value=""
        />
      </View>
      <View
        style={
          {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 16,
              "fontWeight": "600",
              "marginBottom": 8,
            }
          }
        >
          Mật khẩu
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Nhập mật khẩu"
          secureTextEntry={true}
          style={
            {
              "backgroundColor": "white",
              "borderColor": "#E5E5EA",
              "borderRadius": 12,
              "borderWidth": 1,
              "fontSize": 16,
              "paddingHorizontal": 16,
              "paddingVertical": 14,
            }
          }
          testID="password-input"
          value=""
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "marginTop": 8,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#007AFF",
              "fontSize": 14,
              "fontWeight": "500",
            }
          }
        >
          Quên mật khẩu?
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "gap": 16,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#007AFF",
            "borderRadius": 8,
            "justifyContent": "center",
            "minHeight": 44,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 12,
          }
        }
        testID="login-button"
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Đăng nhập
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 8,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#E5E5EA",
              "flex": 1,
              "height": 1,
            }
          }
        />
        <Text
          style={
            {
              "color": "#999",
              "fontSize": 14,
              "marginHorizontal": 16,
            }
          }
        >
          hoặc
        </Text>
        <View
          style={
            {
              "backgroundColor": "#E5E5EA",
              "flex": 1,
              "height": 1,
            }
          }
        />
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#007AFF",
            "borderRadius": 8,
            "borderWidth": 1,
            "justifyContent": "center",
            "minHeight": 44,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 12,
          }
        }
        testID="signup-button"
      >
        <Text
          style={
            {
              "color": "#007AFF",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Tạo tài khoản mới
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "marginTop": 16,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#999",
              "fontSize": 14,
              "textDecorationLine": "underline",
            }
          }
        >
          Bỏ qua ngay bây giờ
        </Text>
      </View>
    </View>
  </View>
</View>
`;
